### ISSUE TITLE
Adding direct URL for interactive and animated figures

### ISSUE THREAD

**Issue by pauldpickell** [ubc-geomatics-textbook/geomatics-textbook#61]
This is related to #60.

The phrasing for interactive captions should be hyperlinked to the exact figure in the hosted HTML output so that someone viewing the PDF build of the textbook can simply click the figure caption and be redirected to the figure as it appears in the online chapter. We can achieve this by appending the chunk label to the base URL for the textbook. For example, the link to chunk `fig:9-sink-fill` (Figure 9.18) in Chapter 9 is [https://ubc-geomatics-textbook.github.io/geomatics-textbook/raster-analysis-and-terrain-modelling.html#fig:9-sink-fill](https://ubc-geomatics-textbook.github.io/geomatics-textbook/raster-analysis-and-terrain-modelling.html#fig:9-sink-fill).

Unfortunately, [bookdown does not natively support embedding URLs in figure captions.](https://community.rstudio.com/t/insert-hyperlink-into-figure-caption/105298/2) So the figure caption would need to use the `<a href>` tag such as:
```
{r 9-sink-fill, fig.cap = fig_cap, out.width= "75%", echo = FALSE}
fig_cap <- paste("Animated process of filling sinks in a DEM. <a href='https://ubc-geomatics-textbook.github.io/geomatics-textbook/raster-analysis-and-terrain-modelling.html#fig:9-sink-fill'>Animated figure can be viewed in the web browser version of the textbook</a>. Pickell, CC-BY-SA-4.0.")
knitr::include_graphics(here::here("images", "09-sink-fill.gif"))
```

However, this enhancement would create a new issue related to the final custom domain name (#15) and final name of the textbook (#24). After some testing, I have discovered the following (not-so-elegant) fix to do this dynamically. It requires an `r` chunk at the top of the chapter `09-raster-analysis-and-terrain-modelling.rmd` (@ect123, not sure if there is a way to set the baseurl as a global variable in bookdown?):

```{r}
baseurl <- "https://ubc-geomatics-textbook.github.io/geomatics-textbook/"
```

Note the trailing `/`. Then we can reference this URL in any figure caption using the `paste0` function:

```
{r 9-sink-fill, fig.cap = fig_cap, out.width= "75%", echo = FALSE}
fig_cap <- paste0("Animated process of filling sinks in a DEM. <a href='",baseurl,"#fig:9-sink-fill'>Animated figure can be viewed in the web browser version of the textbook</a>. Pickell, CC-BY-SA-4.0.")
knitr::include_graphics(here::here("images", "09-sink-fill.gif"))
```

Above yields the following caption in the knitted HTML document (takes a moment to redirect to the location on the page because the Chapter 9 HTML document is very large):

Figure 9.18: Animated process of filling sinks in a DEM. [Animated figure can be viewed in the web browser version of the textbook](https://ubc-geomatics-textbook.github.io/geomatics-textbook/raster-analysis-and-terrain-modelling.html#fig:9-sink-fill). Pickell, CC-BY-SA-4.0.

**Comment by pauldpickell** [comment]
@natasha-sharma123 I just noticed Chapter 8 has a lot of leaflets and other interactive content that was missed when adding these direct URLs to the figure captions. Can you please check all of these? The text in the figure captions needs to be changed to, `Animated figure can be viewed in the web browser version of the textbook` and the appropriate link needs to be added. A lot of the figure captions use City of Vancouver data that also need to be dynamically linked. So there should be two links in a lot of these figures.
